tests/tests/test-eth-fee/test-eth-txn-weights.ts

```diff
@@ -48,6 +48,21 @@ describeDevMoonbeam("Ethereum Weight Accounting", (context) => {
     let blockWeightsUsed = await apiAt.query.system.blockWeight();
     let normalWeight = blockWeightsUsed.normal.refTime.toBigInt();
     expect(normalWeight).to.equal(EXPECTED_WEIGHT);
+
+    // look for the event for our eth txn
+    let extSuccessEvent = null;
+    let wholeBlock = await context.polkadotApi.rpc.chain.getBlock(block.hash);
+    wholeBlock.block.extrinsics.forEach((ext, index) => {
+      if (ext.method.method == "transact" && ext.method.section == "ethereum") {
+        extSuccessEvent = result.events.filter(
```

Comment:
If `extSuccessEvent` can optionally be null, and you are just going to filter for the first result -  we can instead use array.find() to return the first matched event or null, where that block contains an ethereum.transact ext.

Not a huge deal; but for the smoke tests i try to make them as functional as possible.

Reply:
Fixed in 889a285

Reply:
I meant more to remove ‘let extSuccessEvent’ variable from being assigned value in function and instead let it be returned recursively.

e.g. const successEvent = wholeBlock.block.extrinsics.filter().find()